apache-spark-master/Makefile

```diff
@@ -1,24 +1,33 @@
+# Definitions
+spark_versions = 1.5.2 1.6.2 2.0.1
+common_git_commit ?= $(shell git log --pretty=oneline -n 1 -- ../apache-spark-common | cut -f1 -d " ")
+gen_directories = $(spark_versions:=.spark)
 build_tool = runtime-container.DONE
-git_commit ?= $(shell git log --pretty=oneline -n 1 -- ../apache-spark-master | cut -f1 -d " ")
-nametag = quay.io/ucsc_cgl/apache-spark-master:1.5.2--${git_commit}
 
-build: ${build_tool}
+build: $(build_tool)
 
-${build_tool}: Dockerfile run-master.sh
+%.spark: sparkver=$(@:.spark=)
+%.spark: Makefile.template Dockerfile.template
+	mkdir -p $@
+	cp run-master.sh ${@}/
+	sed -e "s:SPARKVER:$(sparkver):g" -e "s:COMMONCOMMIT:$(common_git_commit):g" Dockerfile.template > ${@}/Dockerfile
```

Comment:
May I bring the Dockerfile.py approach that I used for the Toill appliance to the discussion table? I like that it keeps the logic all in one place. While `make` is Turing complete, I'd rather not wrestle with its peculiarities. And believe me, I have wrestled with it a lot in my life.